### ISSUE TITLE
jupyterlab related: better error message if missing jupyterlab-manager?

### ISSUE THREAD

**Issue by hainm** [jupyter-widgets/ipywidgets#1534]
Hi,

I am not sure if I should open this in here or in jupyterlab repo.
Basically I got bellow error in console

```
Error: Object 'jupyter.widget' not found in registry
```

After googling for a while without hope, I realized that I need
```python
jupyter labextension install @jupyter-widgets/jupyterlab-manager
```

Better error message would be super helpful.

**Comment by jasongrout** [comment]
Hmmm...we might be able to trap an error from trying to open the comm on the python side, and if there is an error, print an error about not having the widget frontend installed.

**Comment by jasongrout** [comment]
From http://jupyter-client.readthedocs.io/en/latest/messaging.html#opening-a-comm: "If the target_name key is not found on the receiving side, then it should immediately reply with a comm_close message to avoid an inconsistent state."

@minrk - what do you think about distinguishing between a comm close message that happens because someone explicitly closed the other side, and a close message that happens because the target is not registered on the other side?

**Comment by jasongrout** [comment]
@hainm - it looks like this might take some changes in comm messages, or at least in the comm implementation (if this close reason uses metadata), so bumping to the Future milestone.

**Comment by minrk** [comment]
> what do you think about distinguishing between a comm close message that happens because someone explicitly closed the other side, and a close message that happens because the target is not registered on the other side?

That makes sense. Perhaps a `reason` field?

**Comment by jasongrout** [comment]
> That makes sense. Perhaps a reason field?

Yes. Do you think it should be populated as an enum by a reason code (and have the receiver interpret and display something reasonable) or a free-form user-friendly text field?

**Comment by minrk** [comment]
I suppose it should be something enum-like that's stable and has an official meaning, so kernels/frontends can rely on it. I happen to like strings for this kind of thing a little better than integers. Perhaps `reason` for the strict, stable enum value (e.g. 'notarget') and `message` for a human-readable info that can vary?
